paasta_tools/vitesscluster_tools.py

```diff
@@ -15,6 +17,594 @@
+    throttle_metrics_threshold,
+    tablet_type,
+    region,
+):
+    """
+    Build vttablet config
+    """
+    vttablet_extra_flags = VTTABLET_EXTRA_FLAGS.copy()
+    vttablet_extra_flags[
+        "throttle_metrics_query"
+    ] = f"select max_replication_delay from max_mysql_replication_delay.{throttle_query_table};"
+    vttablet_extra_flags["throttle_metrics_threshold"] = throttle_metrics_threshold
+    vttablet_extra_flags["enforce-tableacl-config"] = "true"
+    vttablet_extra_flags[
+        "table-acl-config"
+    ] = f"/etc/vitess_keyspace_acls/acls_for_{db_name}.json"
+    vttablet_extra_flags["table-acl-config-reload-interval"] = "60s"
+    vttablet_extra_flags["queryserver-config-strict-table-acl"] = "true"
+    vttablet_extra_flags["db-credentials-server"] = "vault"
+    vttablet_extra_flags[
+        "db-credentials-vault-addr"
+    ] = f"https://vault-dre.{region}.yelpcorp.com:8200"
+    vttablet_extra_flags[
+        "db-credentials-vault-path"
+    ] = "secrets/vitess/vt-tablet/vttablet_credentials.json"
+    vttablet_extra_flags[
+        "db-credentials-vault-tls-ca"
+    ] = f"/etc/vault/all_cas/acm-privateca-{region}.crt"
+    vttablet_extra_flags["db-credentials-vault-ttl"] = "60s"
+
+    if tablet_type == "primary":
+        type = "externalmaster"
+    else:
+        type = "externalreplica"
+
+    config = {
+        "cell": cell,
+        "name": f"{db_name}_{tablet_type}",
+        "type": type,
+        "affinity": build_affinity_spec(paasta_pool),
+        "extraLabels": build_extra_labels(paasta_pool, paasta_cluster),
+        "extraEnv": build_extra_env(paasta_cluster),
+        "extraVolumeMounts": [
+            {
+                "mountPath": "/etc/vault/all_cas",
+                "name": "vault-secrets",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "/etc/vitess_keyspace_acls",
+                "name": "acls",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "etc/credentials.yaml",
+                "name": "vttablet-fake-credentials",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "/etc/init_db.sql",
+                "name": "keyspace-fake-init-script",
+                "readOnly": True,
+            },
+        ],
+        "extraVolumes": [
+            {"name": "vault-secrets", "hostPath": {"path": "/nail/etc/vault/all_cas"}},
+            {
+                "name": "acls",
+                "hostPath": {"path": "/nail/srv/configs/vitess_keyspace_acls"},
+            },
+            {"name": "vttablet-fake-credentials", "hostPath": {"path": "/dev/null"}},
+            {"name": "keyspace-fake-init-script", "hostPath": {"path": "/dev/null"}},
+        ],
+        "replicas": 1,
+        "vttablet": {
+            "extraFlags": vttablet_extra_flags,
+            "resources": {
+                "requests": {
+                    "cpu": "100m",
```

Comment:
Same here, have to modify resource specifications based on service metadata in yelpsoa

Reply:
Fixed in https://github.com/Yelp/paasta/pull/3889/commits/6f56e5f7da82df3de44e035247fe865daf409ff4